server/plugins/encryption/tink_state.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encryption
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+)
+
+func (svc *tinkEncryptionService) enable() error {
+	err := svc.callbackOnEnable()
+	if err != nil {
+		return fmt.Errorf("failed enabling TINK encryption: %w", err)
+	}
+	err = svc.updateCiphertextSample()
+	if err != nil {
+		return fmt.Errorf("failed enabling TINK encryption: %w", err)
+	}
+	log.Warn().Msg("encryption enabled")
+	return nil
+}
+
+func (svc *tinkEncryptionService) disable() error {
+	err := svc.callbackOnDisable()
+	if err != nil {
+		return fmt.Errorf("failed disabling TINK encryption: %w", err)
+	}
+	err = svc.deleteCiphertextSample()
+	if err != nil {
+		return fmt.Errorf("failed disabling TINK encryption: %w", err)
+	}
+	log.Warn().Msg("encryption disabled")
+	return nil
+}
+
+func (svc *tinkEncryptionService) rotate() error {
+	newSvc := &tinkEncryptionService{
+		keysetFilePath:    svc.keysetFilePath,
+		primaryKeyID:      "",
+		encryption:        nil,
+		store:             svc.store,
+		keysetFileWatcher: nil,
+		clients:           svc.clients,
+	}
+	err := newSvc.loadKeyset()
+	if err != nil {
+		return fmt.Errorf("failed rotating TINK encryption: %w", err)
```

Comment:
same string is used a lot, we might consider a private const value

Reply:
replaced all of messages with constants. i actually like this approach, it allows to easily find similar/duplicated logs on different abstraction layers